framework/resource_set.go

```diff
@@ -67,7 +66,6 @@ func NewResourceSet(c ResourceSetConfig) (*ResourceSet, error) {
 }
 
 func (r *ResourceSet) InitCtx(ctx context.Context, obj interface{}) (context.Context, error) {
-	ctx = reconciliationcanceledcontext.NewContext(ctx, make(chan struct{}))
 	ctx = updateallowedcontext.NewContext(ctx, make(chan struct{}))
 	ctx = updatenecessarycontext.NewContext(ctx, make(chan struct{}))
```

Comment:
Now resource-related context are set in resource_set.go, and reconciliation-related contexts are set in framework.go.